### ISSUE TITLE
Fusion launchs with toolbars always on top *solved*

### ISSUE THREAD

**Issue by dreamcat4** [cryinkfly/Autodesk-Fusion-360-for-Linux#9]
The problem was first mentioned here:

https://gist.github.com/probonopd/0fab254aa0b6fc371d8db641822bd530#gistcomment-2887370

![Screenshot from 2021-04-23 11-38-01](https://user-images.githubusercontent.com/63074/115859984-c96dde80-a428-11eb-9b7b-1e09663a9a31.png)

So this issue is for to discuss, what is the best way around this problem. And what things we can do about it. Just trying some ideas... will add further details later

**Comment by dreamcat4** [comment]
It does not look like there is any GUI setting in preferences. To just turn off these panels during launch time. That would have been a good workaround.

As it says in main gist, he was clicking on the toolbars to interact with them. However sometimes you click on the toolbar, and it does not work. Like it cannot be clicked. I noticed this problem.

So the best work around ATM is to toggle the 'browser' toolbar + the 'navigation' toolbar. Just hide then show. Using the menu is clunky because it is slow, and also because with this the floating toolbar obscures it.

So I think (for the moment), after every fusion launch. We have best thing do is the following keyboard shortcuts:

```sh
ctrl+alt+b (2x)
ctrl+alt+n (2x)
```

This then fixes the bug for that running session. The bug will always appear back again, after every launch. Every new session.

**Comment by dreamcat4** [comment]
So we should also raise a bug report to Autrodesk themselves. To fix this problem in a future updates. However IDK how to raise the bug report (yet). That will have to be later. Too busy right now to look into it

**Comment by cryinkfly** [comment]
I have a idea 💡 

Create a bash script, that start Fusion 360 and after this comes the commands:

> keyboard shortcuts:
 ctrl+alt+b (2x)
ctrl+alt+n (2x)

**Comment by dreamcat4** [comment]
yes! maybe a tool like autohotkey (but for linux). So the bash script should watch the logfile output from fusion360. And once it detects a certain string in the logs output. To know that fusion has finished booting. Then it can automatically invoke the window (make active window). To send the necessary keyboard shortcuts

**Comment by dreamcat4** [comment]
> ```shell
> ctrl+alt+b (2x)
> ctrl+alt+n (2x)
> ```

+

```sh
> ctrl+alt+a (2x) to toggle comments
```

However now i notice it does not begin to work, until you also click the browser with a mouse click also. So the original commenter in the gist was right.... we still need to do at least 1 'mouse click' at the end. which is not so nice.

Overall it's not terrible. Only takes a moment each launch. However we should also raise the bug report to autodesk developers too. So that they can have some chances maybe to fix it

**Comment by cryinkfly** [comment]
I'll get in touch with Autodesk in the near future. :-)

**Comment by vintagepc** [comment]
This is caused by https://bugs.winehq.org/show_bug.cgi?id=45277

**Comment by cryinkfly** [comment]
@vintagepc I check this out. :-)

**Comment by vintagepc** [comment]
The workaround I've been using (back when F360 didn't crash after login for me) is to configure wine for a virtual desktop, that way these mini-windows are contained and can't overlay other things.

**Comment by cryinkfly** [comment]
I create a way with xdotool for this problem:

> ctrl+alt+b (2x)
ctrl+alt+n (2x)

For more information: https://www.semicomplete.com/projects/xdotool/

**Comment by ManuGithubSteam** [comment]
@cryinkfly  First let me say that this is awesome! I finialy can use fusion without vmware!
I would like to see your script imprved with the virtual desktop to contain these windows so it is a clear boundry between linux windows and Wine/fusion stuff. Also could you share your xdotool solution script ? saves me some time. 
Thanks and very much appreaate your work.

**Comment by cryinkfly** [comment]
Not yet. But I'm working on it. :+1:

**Comment by cryinkfly** [comment]
## I have found a very simple trick to solve this issue! :+1: 

You only have to use the supplied function (Show desktop) from the respective desktop environment so that the toolbars and other elements of Autodesk Fusion 360 are no longer displayed elsewhere. (See video!)

I tested this workaround with XFCE, Gnome and KDE.

https://user-images.githubusercontent.com/79079633/146674225-ff602ffc-7e5e-4b17-abe1-8575f6c5f040.mp4

**Comment by moham96** [comment]
@cryinkfly the workaround doesn't work on manjaro kde

**Comment by cryinkfly** [comment]
Okay, this is interesting. :+1:

**Comment by klack** [comment]
After switching to virtual desktop, many panels disappear underneath the main fusion window (however selection filters work now).  

There doesn't seem to be a keyboard shortcut that fixes the sketch palette after it goes underneath.  So, my solution is to simply stick the sketch palette into the browser window, and resize the main fusion window.

![image](https://user-images.githubusercontent.com/4606571/147615172-86ff4d09-6dad-4138-b2d8-8932c46968bf.png)

**Comment by klack** [comment]
Oh and I forgot to show my thanks and enthusiasm!  Thanks @cryinkfly I may finally be able to switch to Linux for my work PC

**Comment by dreusel** [comment]
I have the same problem, to work around it I usually just minimize the Fusion 360 window when I would normally just mindlessly 'background' windows.
That doesn't solve the issue with menu's in the app itself getting hidden behind the toolbars though.

**Comment by bishboshin** [comment]
Interesting observation: I don't have this issue with Intel GPU (WhiskeyLake-U GT2 [UHD Graphics 620]), only with the Nvidia GPU (GP107M [GeForce GTX 1050 Mobile]).
